### PR TITLE
Feat/binary message parser

### PR DESCRIPTION
**Pull request by jdmnd**

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -18,235 +20,3 @@ export enum STATES {
     PLUGIN_SHUTDOWN,
     LOGGER_SHUTDOWN
 }
-
-export enum EVENT {
```
Comment (yasserf) [inline src/constants.ts]: Where did these events go/

File: src/message/uws/connection-endpoint.ts [hunk 2]
```diff
@@ -144,8 +145,23 @@ export default class UWSConnectionEndpoint extends EventEmitter implements Conne
...
+      let messageBuffer: string | Buffer
+      if (message instanceof ArrayBuffer) {
+        /* we copy the underlying buffer (since a shallow reference won't be safe
+         * outside of the callback)
+         * the copy could be avoided if we make sure not to store references to the
+         * raw buffer within the message
+         */
+        messageBuffer = Buffer.from(message)
+      } else {
+        console.error('received string message', message)
```
Comment (yasserf) [inline src/message/uws/connection-endpoint.ts]: shouldn't this be a log?

**Comment by yasserf** [comment]
Tests fail

**Comment by jdmnd** [comment]
unit tests fixed in https://github.com/deepstreamIO/deepstream.io/pull/841 apart from `protocol/text/test/message-constantsSpec.ts`, since the text messages.ts file is missing some constants

**Comment by yasserf** [comment]
little hard to make an RC release with a PR this big still outstanding.

Can we resolve this so we can make a cut on Monday?

**Pull request merged**